### PR TITLE
[@Powerpages]GitHub copilot request handling

### PR DESCRIPTION
**Pull request by amitjoshi438**
This pull request introduces several changes to the `powerplatform-vscode` codebase to improve the ESLint configuration, update the version, enhance the organization change notifier, and improve the Power Pages chat participant. 

Key changes include:

ESLint Configuration:
* [`.eslintrc.js`](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5R23): Added `ignorePatterns` to ignore the `src/typings` directory during linting. This will help to avoid linting errors from TypeScript definition files that don't need to be linted.

Version Update:
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L5-R5): Updated the version from `1.0.1-dev` to `8.5.0`. This change reflects the new version of the software.

Organization Change Notifier:
* [`src/common/OrgChangeNotifier.ts`](diffhunk://#diff-726acaa16005d3b4352314fee8dbc4879a53dfb2d3e22bd08ce85e7ef9a311a8R54-R56): Added a line to update the global state with the organization ID (`orgID`). This will allow other parts of the application to access the current organization ID from the global state.

Power Pages Chat Participant:
* [`src/common/chat-participants/powerpages/Constants.ts`](diffhunk://#diff-5235feab535c29a1a0cde557c8e53c6914e4f52df10866832bb180495b6afb72R1-R9): Added a constant `NO_PROMPT_MESSAGE` which is a localized string that provides a prompt to get started. This will be used when the user doesn't provide a prompt.
* [`src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts`](diffhunk://#diff-885f58fd5bd7ee590ddae0f7854d056026aff9b576864e07f61887fb7586da29R13-L19): Several changes were made to handle different scenarios in the chat participant class. These include importing additional constants, handling command scenarios, and handling form and list scenarios. [[1]](diffhunk://#diff-885f58fd5bd7ee590ddae0f7854d056026aff9b576864e07f61887fb7586da29R13-L19) [[2]](diffhunk://#diff-885f58fd5bd7ee590ddae0f7854d056026aff9b576864e07f61887fb7586da29L72) [[3]](diffhunk://#diff-885f58fd5bd7ee590ddae0f7854d056026aff9b576864e07f61887fb7586da29R118-R128) [[4]](diffhunk://#diff-885f58fd5bd7ee590ddae0f7854d056026aff9b576864e07f61887fb7586da29L129-R173)

Localization:
* [`loc/translations-export/vscode-powerplatform.xlf`](diffhunk://#diff-62401b58ce452f2686a1cbf6945a0bf5bb33bf35151df029524ced13ada25794R274-R277): Added a new translation unit with an English source text. This will provide a localized string for the prompt to get started.

### CODE REVIEW

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 1]
```diff
@@ -113,25 +112,42 @@ export class PowerPagesChatParticipant {
...
-                    command: ''
-                }
-            };
-        }
+            if (!userPrompt) {
 
-        //TODO: Handle form and list scenarios
-        const llmResponse = await sendApiRequest([{ displayText: userPrompt, code: '' }], { dataverseEntity: '', entityField: '', fieldType: '' }, this.orgID, intelligenceApiToken, '', '', [], this.telemetry, intelligenceEndpoint, geoName);
+                //TODO: String approval is required
+                stream.markdown(NO_PROMPT_MESSAGE);
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:125]: what is this scenario here?

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 2]
```diff
@@ -113,25 +112,42 @@ export class PowerPagesChatParticipant {
...
+
+            const llmResponse = await sendApiRequest([{ displayText: userPrompt, code: '' }], activeFileParams, this.orgID, intelligenceApiToken, '', '', componentInfo, this.telemetry, intelligenceEndpoint, geoName);
 
-        stream.markdown('\n```typescript\n' + llmResponse[0].code + '\n```');
+            llmResponse.forEach((response: { displayText: string | vscode.MarkdownString; code: string; }) => {
+                if (response.displayText) {
+                    stream.markdown(response.displayText);
+                }
+                if (response.code) {
+                    stream.markdown('\n```javascript\n' + response.code + '\n```');
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:145]: how did we finally fix this markdown rendering?

File: src/common/chat-participants/powerpages/PowerPagesChatParticipantConstants.ts [hunk 3]
```diff
@@ -3,8 +3,11 @@
...
 
+import * as vscode from 'vscode';
+
+export const NO_PROMPT_MESSAGE = vscode.l10n.t('Please provide a prompt to get started.\n You can get help with writing code for Power Pages sites in HTML, CSS, and JS languages.');
 export const POWERPAGES_CHAT_PARTICIPANT_ID = 'powerpages';
-export const RESPONSE_AWAITED_MSG = 'Working on it...'
-export const AUTHENTICATION_FAILED_MSG = 'Authentication failed. Please try again.';
-export const COPILOT_NOT_AVAILABLE_MSG = 'Copilot is not available. Please contact your administrator.';
-export const PAC_AUTH_NOT_FOUND = 'Active auth profile is not found or has expired. Please try again.';
+export const RESPONSE_AWAITED_MSG = vscode.l10n.t('Working on it...');
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipantConstants.ts:10]: run complete gulp ci once - this should be added to bundle json file

File: src/common/chat-participants/powerpages/PowerPagesChatParticipantUtils.ts [hunk 4]
```diff
@@ -16,3 +20,29 @@ export async function getEndpoint(
...
+/*
+    * Get component info for the active file
+    * @returns componentInfo - Entity details for active file (form or list)
+*/
+export async function getComponentInfo(telemetry: ITelemetry, orgUrl: string | undefined, activeFileParams: IActiveFileParams): Promise<string[]> {
+
+    let metadataInfo = { entityName: '', formName: '' };
+    let componentInfo: string[] = [];
+
+    if (activeFileParams.dataverseEntity == ADX_ENTITYFORM || activeFileParams.dataverseEntity == ADX_ENTITYLIST) {
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipantUtils.ts:33]: lets move this to a utility function - that will help when we expand to other entities

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts (hunk 1 above)
Comment (amitjoshi438) [reply src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:125]: It was for dataverse integration for form and list

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts (hunk 2 above)
Comment (amitjoshi438) [reply src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:145]: Adding the newline character worked for our responses. The example I was trying must be missing something on same lines.

**Review by gshivi — APPROVED** [review]

**Pull request merged**